File: src/optical_element/MATLAB/Furion_roty.m
function R = Furion_roty( t )
% rotation angle t about the y-axis ;left-hand column vector;
% right-hand row vector;
%************************************************************************
% ID:     21062904
% ***********************************************************************
% Input:  angle , unit radian
% ***********************************************************************
% Ouput:  transform matrix
% ***********************************************************************
%
% Modification history
%   2021-6-29 add annotation;
%************************************************************************
ct=cos(t);
st=sin(t);
R=[ 
    ct,0,-st;
    0,1,0;
    st,0,ct;
    ];
end

File: src/optical_element/furion_roty.cpp
#include "Furion_Roty.h"

using namespace Furion_NS;

Furion_Roty::Furion_Roty()
{

}

Furion_Roty::~Furion_Roty()
{
    
}

void Furion_Roty::furion_roty(double t, std::vector<double>& matrix)
{
    double ct = cos(t);
    double st = sin(t);
    double data[9] = {ct, 0, -st, 0, 1, 0, st, 0, ct};

    for (int i = 0; i < 9; i++)
    {
        matrix[i] = data[i];
    }
}

